src/crypto/openssl/cose_sign.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/openssl/cose_sign.h"
+
+#include "ccf/ds/logger.h"
+
+#include <openssl/evp.h>
+#include <t_cose/t_cose_sign1_sign.h>
+
+namespace
+{
+  constexpr int64_t COSE_HEADER_PARAM_ALG =
```

Comment:
I suspect we will probably want to start a `cose/sign.cpp` and `cose/constants.h` scheme quite soon, but this is fine right now.